src/common/exif.cc

```diff
@@ -1110,6 +1120,16 @@ int dt_exif_write_blob(uint8_t *blob, uint32_t size, const char *path, const int
       dt_remove_exif_keys(imgExifData, keys, n_keys);
     }
 
+    // remove subimage* trees, related to thumbnails or HDR usually
+    for(Exiv2::ExifData::iterator i = imgExifData.begin(); i != imgExifData.end();)
+    {
+      std::string needle = "Exif.SubImage";
+      if(i->key().compare(0, needle.length(), needle) == 0)
```

Comment:
Hm, now i'm not sure whether this too should be moved into `dt_exif_read_blob()` (but not into ` if(!dng_mode){}`)

Why? Because this function first reads the final written image, and then overrides the tags from `blob`

I.e. i'm not sure that it is guaranteed there won't ever be valid `Exif.SubImage*` tags here.

`dt_exif_read_blob()` only reads the original input file, so it should be safe (safer?) there...